code/modules/clothing/chameleon.dm

```diff
@@ -681,3 +681,47 @@
 /obj/item/clothing/neck/chameleon/broken/Initialize()
 	. = ..()
 	chameleon_action.emp_randomise(INFINITY)
+
+/datum/action/item_action/chameleon/change/skillchip
+	/// Skillchip this this chameleon action is imitating.
+	var/obj/item/skillchip/skillchip_mimic
+	var/obj/item/skillchip/usable_at_world_time = 0
+	var/cooldown = 5 MINUTES
+
+/datum/action/item_action/chameleon/change/skillchip/initialize_disguises()
+	. = ..()
```

Comment:
I'd prefer if what skillchips are availible here were made explicit by some flag and/or selected subtypes. Since i don't expect all skillchips to be easily available in the future.

Reply:
I'm going to implement a middleground solution and add a flag check alongside this method.

The problem is - I want to exclude certain type paths from the Chameleon item list such as the parent type /obj/item/skillchip.

I can simply add a flag and stop using this blacklist, but then every single child of /obj/item/skillchip HAS to override the flags.

The reason /obj/item/skillchip would have the chameleon incompatibility flag set is not because it should be the default behaviour of skill chips, but because /obj/item/skillchip isn't a REAL skillchip.

Anyone who forgot to clear the flags when inheriting /obj/item/skillchip would have unexpected functionality in their chips. It's effectively an open season for introducing bugs. I suspect this is part of why chameleon items use the blacklist method that they do.

As a result, I'm doing it both ways. I'm going to hardcode "abstract" item paths in chameleon_blacklist however when building the item list, it also checks for the presence of the appropriate flag to exclude the item too.

The end system will be as follows: Real skillchips that want to be chameleon blacklisted can use the flag. Abstract skillchips that exist just as parents and should never actually appear in-game can have their paths added to the blacklist without forcing ALL their children to override thier flags.

Reply:
what if you just used subtypesof instead of typesof or whatever you're using to build the list? the base type should be the easiest to remove.